### PR TITLE
[2/N] Non-Tensor: Scalar Support: Add scalar to the cache for eager-through-torch.compile

### PR DESCRIPTION
**Pull request by EikanWang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #126517
* #125897
* #125831
* #125819
* #125308
* #124926
* __->__ #124070

Add scalar information to the kernel configuration.

#### Additional Context
Currently, the input parameters are orchestrated by input order in the kernel configuration and loaded/mapped to the kernel at runtime. For example, the cache order of the input parameters of `torch.add(a, b, alpha=2.0)` is `a' first, followed by `b` and then `alpha`. The same order is for cache loading.

However, the orchestration mechanism does not support kwargs because the order of kwargs is useless. For example, the `out` of `aten::gelu.out(Tensor self, *, str approximate='none', Tensor(a!) out) -> Tensor(a!)` may be before `approximate`. We will support it with subsequent PRs. 

cc @voznesenskym @penguinwu @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/utils.py [hunk 1]
```diff
@@ -1640,8 +1658,14 @@ def aoti_compile_with_persistent_cache(
                 else:
                     meta_info_item["device_index"] = f"{input_tensor.device.index}"
                 meta_info_item["dtype"] = f"{input_tensor.dtype}"
-                meta_info_item["sizes"] = f"{list(input_tensor.size())}"
-                meta_info_item["strides"] = f"{list(input_tensor.stride())}"
+                if input_tensor.ndim == 0:
+                    # Scalar tensor
+                    meta_info_item["sizes"] = "[]"
+                    meta_info_item["strides"] = "[]"
+                    meta_info_item["scalar_value"] = f"{input_tensor.item()}"
```
Comment (jansel) [inline torch/_inductor/utils.py:1665]: Don't we only want to do this if the original value came from an int/float/bool?  If the user provides a 0D tensor I would expect it not to specialize.

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_inductor/utils.py (hunk 1 above)
Comment (EikanWang) [reply torch/_inductor/utils.py:1695]: Make sense. We should not specialize 0D tensor. I will refine the PR. Thanks for the comments.

**Review by jgong5 — APPROVED** [review]
LGTM except for some small nits.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**